### PR TITLE
Unit Testing for mentorMatchUtils.js (TC #1 Testing)

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- **What does this PR do?**  
  Adds `mentorMatchUtils.test.js` with full unit-tests for every helper in **mentorMatchUtils.js** and wires up a test script.  
- **Why is it necessary or valuable?**  
  Makes sure mentor-matching logic is bullet-proof. Stops hidden bugs from sneaking in.  

---

# Milestones / Related Work  
- **Milestone / version:** [4.14 Unit Testing (TC #1)](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.7wp2edl8cz1p)

---

# Resources and References  
- Vitest docs – https://vitest.dev  
- Istanbul coverage guide – https://istanbul.js.org  

---

# Test Plan  
- Ran `npm run test: all green
image

- **Seed / mock data:** simple user/mentor objects in the test file.

## Edge Cases Tested  
- User with no matching skills / interests.  
- Experience years > max cap.  
- Like map empty vs. populated.  
- Cosine similarity when one vector is all zeros.

---

# Additional Notes  
- **Reviewer focus:** naming clarity in tests, extra edge cases you think we missed.

### CODE REVIEW

File: frontend/src/__tests__/mentorMatchUtils.test.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	it("encodeMeeting one-hot encodes option", () => {
+		expect(encodeMeeting(user)).toEqual([1, 0, 0]); // Zoom chosen
+	});
+});
+
+// vector + similarity
+describe("vectorizeUser + cosineSimilarity", () => {
+	it("vectorizeUser builds full weighted vector", () => {
+		const vec = vectorizeUser(user, SKILLS, INTERESTS, W);
+		const expectedLen = SKILLS.length + INTERESTS.length + 1 + 1 + 3; // 3+3+1+1+3
```
Comment (alegleason) [inline frontend/src/__tests__/mentorMatchUtils.test.js:74]: Can you explain the rational behind 1 + 1 + 3? Maybe having them as `const`s is worth it.

Comment (Calebc4554) [reply frontend/src/__tests__/mentorMatchUtils.test.js:74]: yea good point! the 1 + 1 + 3 is for duration, attendees, and location categories (loc:academic, loc:virtual, loc:other). I can pull those into consts to make it clearer.

Comment (Calebc4554) [reply frontend/src/__tests__/mentorMatchUtils.test.js:74]: yea for sure the 1 + 1 + 3 is for, 1 = AI interest, 1 = experience years, 3 = meeting pref (Zoom, Hybrid, In Person). I can put that in const tho
